src/domain/ecoverse/ecoverse.service.ts

```diff
@@ -73,19 +72,32 @@ export class EcoverseService {
       ecoverse.partners = [];
     }
 
-    if (!ecoverse.tagset) {
-      this.tagsetService.initialiseMembers(ecoverse.tagset);
-    }
+    // Initialise contained singletons
+    this.tagsetService.initialiseMembers(ecoverse.tagset);
+    this.contextService.initialiseMembers(ecoverse.context);
 
     return ecoverse;
   }
 
   async getEcoverse(): Promise<IEcoverse> {
     try {
-      const ecoverse = await this.ecoverseRepository.findOneOrFail();
+      const ecoverseArray = await this.ecoverseRepository.find();
+      const ecoverseCount = ecoverseArray.length;
+      if (ecoverseCount == 0) {
+        console.log('No ecoverse found, creating empty ecoverse...');
+        // Create a new ecoverse
+        const ecoverse = new Ecoverse();
```

Comment:
this method needs refactoring - it's misleading

Reply:
When the boostrap module is there then this code should go completely